src/gui/qgsattributeform.cpp

```diff
@@ -2483,23 +2483,27 @@ void QgsAttributeForm::updateDefaultValueDependencies()
     QgsEditorWidgetWrapper *eww = qobject_cast<QgsEditorWidgetWrapper *>( ww );
     if ( eww )
     {
-      QgsExpression exp( eww->field().defaultValueDefinition().expression() );
-      const QSet<QString> referencedColumns = exp.referencedColumns();
-      for ( const QString &referencedColumn : referencedColumns )
+      const QgsDefaultValue defaultValueDefinition = eww->field().defaultValueDefinition();
+      if ( defaultValueDefinition.applyOnUpdate() )
```

Comment:
I'm not sure if this is what the user expects: What does "apply on update" mean? 1. Possibility: Apply on update of the feature (means editing and not adding it) or 2. Possibilty: on update another field (on adding or editing the feature)? 

Currently the first possibility is integrated. Means it checks this only on adding a new feature here https://github.com/qgis/QGIS/blob/master/src/gui/qgsattributeform.cpp#L510 

It's what it should do in my opinion. But if I'm wrong, it should anyway handle it on the line 510, no?

Reply:
> I'm not sure if this is what the user expects: What does "apply on update" mean? 1. Possibility: Apply on update of the feature (means editing and not adding it) or 2. Possibilty: on update another field (on adding or editing the feature)?

I agree this could be clearer. The original idea was clearly 1.

But, @nyalldawson had the opposite perception in the following discussion https://github.com/qgis/QGIS/pull/35030#issuecomment-598343631, so currently we have a warning on the dialog (see below) that indicates this is not supposed to be working.

![image](https://user-images.githubusercontent.com/588407/85504913-553a9280-b5ed-11ea-8c41-b1b85731af28.png)

Reply:
I remember that discussion. Fair enough but wouldn't it be clearer with an option like: "Apply default value on attribute updates"? Since the values should be updated when this option is checked.

And then catch it here instead https://github.com/qgis/QGIS/blob/master/src/gui/qgsattributeform.cpp#L510

When then the "Apply default value on update" is checked we would have the issue  #37359 still, no? So we'll need a follow up fix.

Reply:
> but wouldn't it be clearer with an option like: "Apply default value on attribute updates"? Since the values should be updated when this option is checked.

That would be a possibility but would add a new option and therefore not be very backport friendly?
This patch aims to fix a inconsistency between the warning and the behavior. The question is if the warning is wrong or the behavior is wrong. If we change the behavior, we will get a band aid for the particular issue #37359. But I'm open to kill the warning instead as well.

> When then the "Apply default value on update" is checked we would have the issue #37359 still, no? So we'll need a follow up fix.

Correct, but this is a completely separate fix, where the snapping scope should be preserved in the attribute form.